### ISSUE TITLE
Implement scriptlet injection

### ISSUE THREAD

**Issue by pes10k** [brave/brave-ios#2684]
Similar to https://github.com/brave/brave-ios/issues/2683, scriptlets could be injected into documents without needing to use the iOS content blocking system, and much of the [content script code](https://github.com/brave/brave-core/blob/master/components/brave_extension/extension/brave_extension/content_cosmetic.ts) from brave-core could be reused.

**Comment by bsclifton** [comment]
@pes10k did you have more information about this? Is this a proposed replacement of the content blocker files we use? (ex: the final output of slim list; generated with adblock-rust now, but was generated with AB2CB before)

**Comment by pes10k** [comment]
In desktop this is handled by adblock-rs and the content script in the brave extension 
In android its handled by adblock-rs and custom code @SergeyZhukovsky is mantaining

I think the adblock-rs part could be easily reused, and iOS could (like android) just inject something like https://github.com/brave/brave-core/blob/master/components/brave_extension/extension/brave_extension/content_cosmetic.ts#L180 into pages.  The only part that'd need to be re-implemented is the page <-> browser communication channel

**Comment by Emasoft** [comment]
Brave should allow the user to browse tampermonkey scripts from sites like  https://greasyfork.org/ and automatically install them, like the GEAR Browser for iOS.

**Comment by pes10k** [comment]
@Emasoft thats a pretty different issue than whats being described here (which is about the scriplets already being used and deployed on desktop and android).  If you'd like the ability to add more scriptlets from a source other than brave's common lists, please create another issue

Just a side note too that adding custom more stuff to the browser makes you more identifying, so if there are features you think would be generally useful, please please suggest them. We're far more eager to consider features that change how the browser works for everyone, than to add more customization points, since more customization harms privacy, so we need to be more cautious about privacy / utlity trade off there

**Comment by kylehickinson** [comment]
Blocked by https://github.com/brave/brave-ios/issues/3683

**Comment by anthonypkeane** [comment]
For a spec, please search for "Cosmetic Filters and Scriptlet Injections" in GDrive

**Comment by Uni-verse** [comment]
Verified in https://github.com/brave/brave-ios/issues/2683#issuecomment-1085243900
